code/game/machinery/mind_machine.dm

```diff
@@ -0,0 +1,846 @@
+	if(href_list["scan_start"])
+		if(!currentlySwapping)
+			scanOccupants()
+			return TRUE
+		else
+			errorMessage = "Currently active: Please wait"
+			return TRUE
+	if(href_list["unlock_pods"])
+		if(((!connectOne.contents.len) || (!connectTwo.contents.len)) && (currentlySwapping))
+			unlockPods() //Safeguard against the pod getting trapped in infinite scan
+			return TRUE
+		if(lockedPods == FALSE)
+			errorMessage = "Pods already unlocked"
+			return TRUE
+		if(currentlySwapping)
+			errorMessage = "Currently active: Please wait"
+			return TRUE
+		else
+			spawn(5)
+				unlockPods()
+				return TRUE
+	if(href_list["eject_pods"])
+		if((!currentlySwapping) && (lockedPods == FALSE))
+			spawn(5)
+				ejectPods()
+				return TRUE
+		if(lockedPods == TRUE)
+			errorMessage = "Pods are locked"
+			return TRUE
+	if(href_list["mind_swap"])
+		if((occupantScan) && (!currentlySwapping))
+			spawn(5)
+				swapOccupants()
+				return TRUE
+		if(!occupantScan)
+			errorMessage = "Require scan"
+			return TRUE
+		if(currentlySwapping)
+			errorMessage = "Currently active: Please wait"
+			return TRUE
+	if(href_list["delay_swap"])
+		if(!currentlySwapping)
+			currentlySwapping = TRUE
+			spawn(5)
+				delayedSwap()
+				return TRUE
+
+//////Scan and Swap, other UI procs//////////////
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/unlockPods()
+	occupantScan = FALSE
+	mindTypeOne = "None"
+	mindTypeTwo = "None"
+	occupantNameOne = "None"
+	occupantNameTwo = "None"
+	occupantStatOne = "None"
+	occupantStatTwo = "None"
+	connectOne.icon_state = "mind_pod_open"
+	connectTwo.icon_state = "mind_pod_open"
+	playsound(connectTwo, 'sound/machines/door_unbolt.ogg', 35, 1)
+	playsound(connectOne, 'sound/machines/door_unbolt.ogg', 35, 1)
+	if(badSwap == FALSE)
+		flick("mind_pod_opening", connectOne)
+		flick("mind_pod_opening", connectTwo)
+	else
+		flick("mind_pod_thefly", connectOne)
+		flick("mind_pod_thefly", connectTwo)
+		badSwap = FALSE
+	lockedPods = FALSE
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/ejectPods()
+	connectOne.go_out()
+	connectTwo.go_out()
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/delayedSwap()
+	if(!occupantScan)
+		errorMessage = "Proceed with caution: supervisor advised"
+		sleep(5 SECONDS)
+		currentlySwapping = FALSE
+		if((occupantOne) && (occupantTwo))
+			scanOccupants()
+			if(prob(8-manipRating)) //A tiny bit risky with no upgrades, 0 risk with full xenoarch manipulators.
+				malfSwap = TRUE
+			swapOccupants()
+			unlockPods()
+		else
+			errorMessage = "Missing occupant"
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/scanOccupants(var/mob/S)
+	if((!locate(occupantOne,connectOne.contents)) || (!locate(occupantTwo, connectTwo.contents)))
+		errorMessage = "Missing occupant"
+		occupantOne = null
+		occupantTwo = null
+		return
+	currentlySwapping = TRUE
+	lockedPods = TRUE
+	if((connectOne.icon_state != "mind_pod_closed") && (connectTwo.icon_state != "mind_pod_closed"))
```

Comment:
Checking icon states to check "were the pods already locked" is a really bad idea. You should keep track of a var on them or something.